### PR TITLE
adding a mutex to firebase to prevent race conditions during fb.Watch

### LINKED ISSUE #10: Race Conditions with fb.Watch
Running `go test -race .` you get something like

https://travis-ci.org/CloudCom/firego/jobs/64910115#L158

### PR DESCRIPTION
**Pull request by zabawaba99**
Resolves #10

### CODE REVIEW

File: watch.go [hunk 1]
```diff
@@ -71,7 +85,11 @@ func (fb *Firebase) Watch(notifications chan Event) error {
 		// if we're told to stop, close the response Body
 		go func() {
 			<-fb.stopWatching
+
+			fb.watchMtx.Lock()
 			closedManually = true
+			fb.watchMtx.Unlock()
```
Comment (MrGossett) [inline watch.go]: `closedManually` is not a property of `fb` and is scoped locally within this func. I'm not sure it's appropriate to use `fb.watchMtx` here. Perhaps `closedManually` itself could be replaced by something like this:

``` go
type ConcurrentBool struct {
    b bool
    sync.RWMutex
}

func (b *ConcurrentBool) Value() bool {
    b.RLock()
    value := b.b
    b.RUnlock()
    return value
}

func (b *ConcurrentBool) Set(v bool) {
    b.Lock()
    b.b = v
    b.Unlock()
}
```

... or something using `sync/atomic`

Comment (zabawaba99) [inline watch.go]: Fair enough. However, I don't think it's worth building a whole new type just for this one function. We also can't use `sync/atomic` unless we use 'unsafe.Pointer` which i'd like to stay away from.

I'll just add another mutex inside that function.

**Comment by MrGossett** [comment]
build failed with

```
Bad response status from coveralls: 422 - {"message":"Couldn't find a repository matching this job.","error":true}

The command "$HOME/gopath/bin/goveralls -service=travis-ci -repotoken=$COVERALLS -v" exited with 1.
```

**Comment by zabawaba99** [comment]
Build failure has been address

**Comment by MrGossett** [comment]
LGTM. :rocket:

**Pull request merged**